features/netsocket/PPP.h

```diff
@@ -0,0 +1,165 @@
+     */
+    //typedef void (*ppp_link_input_fn)(void *data, net_stack_mem_buf_t *buf);
+    typedef mbed::Callback<void (net_stack_mem_buf_t *buf)> ppp_link_input_cb_t;
+
+    /**
+     * Callback to be registered with PPP interface and to be called for link status changes
+     *
+     * @param  up   Link status
+     */
+    //typedef void (*ppp_link_state_change_fn)(void *data, bool up);
+    typedef mbed::Callback<void (bool up)> ppp_link_state_change_cb_t;
+
+    /**
+     * Return maximum transmission unit
+     *
+     * @return     MTU in bytes
+     */
+    virtual uint32_t get_mtu_size() = 0;
+
+    /**
+     * Gets memory buffer alignment preference
+     *
+     * Gets preferred memory buffer alignment of the cellular device.
+     * @return         Memory alignment requirement in bytes
+     */
+    virtual uint32_t get_align_preference() const = 0;
+
+    /**
+     * Return interface name
+     *
+     * @param name Pointer to where the name should be written
+     * @param size Maximum number of characters to copy
+     */
+    virtual void get_ifname(char *name, uint8_t size) const = 0;
+
+    /**
+     * Sends the packet over the link
+     *
+     * That cannot be called from an interrupt context.
+     *
+     * @param buf  Packet to be sent
+     * @return     True if the packet was sent, false otherwise
+     */
+    virtual bool link_out(net_stack_mem_buf_t *buf, nsapi_ip_stack_t ip_stack) = 0;
+
+    /**
+     * Initializes the PPP
+     *
+     * @return True on success, False in case of an error.
+     */
+    virtual bool power_up() = 0;
+
+    /**
+     * Deinitializes the PPP
+     *
+     */
+    virtual void power_down() = 0;
+
+    /**
+     * Sets a callback that needs to be called for packets received for that interface
+     *
+     * @param input_cb Function to be register as a callback
+     */
+    virtual void set_link_input_cb(ppp_link_input_cb_t input_cb) = 0;
+
+    /**
+     * Sets a callback that needs to be called on link status changes for given interface
+     *
+     * @param state_cb Function to be register as a callback
+     */
+    virtual void set_link_state_cb(ppp_link_state_change_cb_t state_cb) = 0;
+
+    /** Sets memory manager that is used to handle memory buffers
+     *
+     * @param mem_mngr Pointer to memory manager
+     */
+    virtual void set_memory_manager(NetStackMemoryManager &mem_mngr) = 0;
+
+    /** Sets file stream used to communicate with modem
+     *
+     * @param stream Pointer to file handle
+     */
+    virtual void set_stream(mbed::FileHandle *stream) = 0;
+
+    /** Sets IP protocol versions of IP stack
+     *
+     * @param ip_stack IP protocol version
+     */
+    virtual void set_ip_stack(nsapi_ip_stack_t ip_stack) = 0;
+
+    /** Sets user name and password for PPP protocol
+     *
+     * @param uname User name
+     * @param passwrd Password
+     */
+    virtual void set_uname_password(const char *uname, const char *password) = 0;
```

Comment:
Cellular has used `set_credentials` name for these.

Reply:
Renamed.